testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/config/smallrye/converter/MicroProfileConfigConvertersTestCase.java

```diff
@@ -0,0 +1,62 @@
+package org.wildfly.test.integration.microprofile.config.smallrye.converter;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.microprofile.config.smallrye.AssertUtils;
+
+import java.net.URL;
+
+import static org.wildfly.test.integration.microprofile.config.smallrye.HttpUtils.getContent;
+
+/**
+ * @author <a href="mailto:mjurc@redhat.com">Michal Jurc</a> (c) 2018 Red Hat, Inc.
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(SetupTask.class)
+public class MicroProfileConfigConvertersTestCase {
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "MicroProfileConfigConvertersTestCase.war")
+                .addClasses(AutomaticallyConvertibleByConstructor.class, AutomaticallyConvertibleByOf.class,
+                        AutomaticallyConvertibleByParse.class, AutomaticallyConvertibleByValueOf.class, TestApplication.class,
+                        TestApplication.Resource.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsWebInfResource(TestApplication.class.getPackage(), "jboss-deployment-structure.xml",
```

Comment:
it is necessary? If the converters are added to the archive, they should be accessible to convert type (otherwise, that's a bug)

Reply:
Basically to explain what I think I am doing in this test:
* Create module in `SetupTask` that exports `Return101Converter`, `Return102Converter`, `HighPriorityStringConverter1`, `HighPriorityStringConverter2` service implementations. 
* Create deployment in `MicroProfileConfigConvertersTestCase` that depends on the aforementioned module. The deployment also has some classes that should be automatically converted without use of `Converter` services (`AutomaticallyCovertibleBy*`).